yarn-project/simulator/src/public/enqueued_calls_processor.ts

```diff
@@ -265,17 +331,28 @@ export class EnqueuedCallsProcessor {
         );
         throw enqueuedCallResult.revertReason;
       }
+      await txStateManager.mergeStateForEnqueuedCall(
```

Comment:
Note to self: I'm wondering why we can't have a generic accept/reject and we need a special one for enqueued call?

Reply:
Because this merge function traces the enqueued call under the hood, adding it to hints & vector that ultimately becomes public inputs

Reply:
Right now, this is how we handle all "merges of forked state". We merge the forked state, and we trace the thing that caused the fork. It's not pretty....

Maybe the merge and trace should be separate functions? So we first do mergeForkedState which is always the same no matter what. And then we call a tailored trace function?